### PR TITLE
use tsup to build library

### PR DESCRIPTION
**Pull request by evertbouw**

### CODE REVIEW

File: tsup.config.ts
Comment (jayphelps) [inline tsup.config.ts:1]: I hadn't heard of tsup before today. I'm weary switching to something I'm completely unfamiliar with, but if you think it's ready for prime time and won't be abandoned any time soon, I'm down with it.

**Review by jayphelps — APPROVED** [review]
Thank you!! LGTM. Feel free to merge when ready.

File: tsup.config.ts
Comment (evertbouw) [reply tsup.config.ts:1]: I trust that acemarke got it right https://blog.isquaredsoftware.com/2023/08/esm-modernization-lessons/
if it works for redux it'll work for us™️

Comment (jayphelps) [reply tsup.config.ts:1]: :shipit:

Comment (markerikson) [reply tsup.config.ts:1]: 🤣 🤣 🤣 

😭 😭 😭

**Pull request merged**